### PR TITLE
Fixed the crash that might occur when stopping the raft service

### PR DESCRIPTION
**Pull request by monadbobo**
1. When the ThriftClientManager object needs to be shared, if the EventBase object was released, the corresponding host was removed from the ThriftClientManager.
2. fixed the compile warning by Clang.

### CODE REVIEW

**Comment by monadbobo** [comment]
close #1077

File: src/common/thrift/ThriftClientManager.h [hunk 1]
```diff
@@ -35,7 +35,7 @@ class ThriftClientManager final {
         std::shared_ptr<ClientType>                 // Async thrift client
     >;
 
-    folly::ThreadLocal<ClientMap> clientMap_;
+    ClientMap                     clientMap_;
```
Comment (dangleptr) [inline src/common/thrift/ThriftClientManager.h]: Why not use static clientMap_ directly?

Comment (monadbobo) [reply src/common/thrift/ThriftClientManager.h]: Whether to use static here is the same, because we use the ThriftClientManager object as thread_local.

Comment (dangleptr) [reply src/common/thrift/ThriftClientManager.h]: I mean why not use “static folly::ThreadLocal clientMap”

Comment (monadbobo) [reply src/common/thrift/ThriftClientManager.h]: If we used "static folly::ThreadLocal clientMap", all used clients (meta/storage) will share a clientMap. For example, adminclient and storageclient will likely use the same StorageServiceAsyncClient object. Is this necessary?

Comment (dangleptr) [reply src/common/thrift/ThriftClientManager.h]: I think it's ok for connections.   The same thread share the same connection.  wdyt?

**Pull request closed without merging**